src/coreclr/src/hosts/unixcorerun/CMakeLists.txt

```diff
@@ -30,4 +30,8 @@ if(NOT CLR_CMAKE_TARGET_ANDROID)
   )
 endif()
 
+if(CLR_CMAKE_TARGET_ARCH_I386 OR CLR_CMAKE_TARGET_ARCH_ARM)
+    add_definitions(-D_FILE_OFFSET_BITS=64)
```

Comment:
I'm surprised this wasn't set before.  This should actually be set globally, but then we'd need to check if every function calling into filesystem APIs aren't using unnamed primitive types... and verify if interop scenarios will be using the correct types.

Reply:
We do set it for the PAL. Setting it in the corerun was clearly an omission. It seems that we should set it in other hosts (in the installer) as well.

As for this PR, we should set it in a single location - in the src/coreclr/src/hosts/CMakeLists.txt instead. And a nit - it doesn't need to be set for 32 bit architectures only, it doesn't hurt to set it unconditionally.

Reply:
@NextTurn I am sorry for missing this PR when you have originally created it, I could have given you this feedback much earlier.

Reply:
@janvorli It's OK. Thanks for reviewing this!